public/app/features/alerting/unified/components/amroutes/AmRoutesTable.tsx

```diff
@@ -6,34 +6,57 @@ import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '..
 import { AmRoutesExpandedForm } from './AmRoutesExpandedForm';
 import { AmRoutesExpandedRead } from './AmRoutesExpandedRead';
 import { Matchers } from '../silences/Matchers';
-import { matcherFieldToMatcher } from '../../utils/alertmanager';
+import { matcherFieldToMatcher, parseMatchers } from '../../utils/alertmanager';
+import { intersectionWith, isEqual } from 'lodash';
+import { EmptyArea } from '../EmptyArea';
 
 export interface AmRoutesTableProps {
   isAddMode: boolean;
   onChange: (routes: FormAmRoute[]) => void;
   onCancelAdd: () => void;
   receivers: AmRouteReceiver[];
   routes: FormAmRoute[];
+  filters?: { queryString?: string; contactPoint?: string };
   readOnly?: boolean;
 }
 
 type RouteTableColumnProps = DynamicTableColumnProps<FormAmRoute>;
 type RouteTableItemProps = DynamicTableItemProps<FormAmRoute>;
 
+const getFilteredRoutes = (routes: FormAmRoute[], labelMatcherQuery?: string, contactPointQuery?: string) => {
```

Comment:
I wonder if this should be a util or hook so we can write a test for it? Thoughts?

Reply:
Good point. I was wondering what is a good approach here. I feel this cannot be used in more places than this, so moving it to `utils` or `hooks` seems a bit excessive. I can move it to a hook and maybe put it in a separate file just next to the `AmRoutesTable.tsx`?

Reply:
I don't have any strong opinions on this one – I feel like it's fine where it is since it deals with `FormAmRoute[]` which is a data structure that is only handled here.

This one could need a test though :D

Reply:
I added a few tests